kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -300,21 +290,24 @@ BlockId DBAdapter::getLatestBlock() const {
     LOG_TRACE(logger_, "Latest block ID " << blockId);
     return blockId;
   }
-  // No state transfer blocks in the system. Fallback to getLastReachableBlock() .
-  return getLastReachableBlock();
+  // No state transfer blocks in the system. Fallback to getLastReachableBlockId() .
+  return getLastReachableBlockId();
 }
 
 Sliver DBAdapter::createBlockNode(const SetOfKeyValuePairs &updates, BlockId blockId) const {
   // Make sure the digest is zero-initialized by using {} initialization.
   auto parentBlockDigest = StateTransferDigest{};
   if (blockId > 1) {
-    Sliver parentBlock;
-    const auto status = getBlockById(blockId - 1, parentBlock);
-    if (!status.isOK() || parentBlock.empty()) {
-      LOG_FATAL(logger_, "createBlockNode: no block or block data for parent block ID " << blockId - 1);
+    try {
+      const auto parentBlock = getRawBlock(blockId - 1);
+      computeBlockDigest(blockId - 1, parentBlock.data(), parentBlock.length(), &parentBlockDigest);
+    } catch (const std::exception &e) {
+      LOG_FATAL(logger_, "Failed to get parent block ID = " + std::to_string(blockId - 1) + ", reason: " + e.what());
+      std::exit(1);
```

Comment:
I prefer to have exit(1) in one place where the exceptions are caught

Reply:
I agree - in that particular case we can just throw, doing that.